metadata-ingestion/src/datahub/sdk/_shared.py

```diff
@@ -204,6 +269,29 @@ def _set_container(self, container: Optional[ContainerInputType]) -> None:
                     urn=container_urn,
                 ),
             ]
+        elif isinstance(container, list):
+            parsed_path = [self._maybe_parse_as_urn(entry) for entry in container]
+
+            # Use the last container in the path as the container urn.
+            container_urns = [
+                urn.urn() for urn in parsed_path if isinstance(urn, ContainerUrn)
+            ]
+            container_urn = container_urns[-1] if container_urns else None
+
+            browse_path = [
+                (
+                    models.BrowsePathEntryClass(
+                        id=str(entry),
+                        urn=str(entry),
```

Comment:
```suggestion
                        id=entry.urn(),
                        urn=entry.urn(),
```

Although nowadays `__str__` is implemented with `urn`, I think it's better to rely on the contract of this method

```
    def urn(self) -> str:
        """Get the string representation of the urn."""
```

Reply:
I agree that some proper method is better than `str()`. However I think naming the method `.urn()` was a bad choice since it's misleading that .urn returns a string, and not an urn type. 

So longer term I'd like to add a `as_str` method and start using that everywhere instead. For now, I'll just leave this as-is